common/transport/amqp/src/amqp_cbs.ts

```diff
@@ -217,22 +225,25 @@ export class ClaimsBasedSecurityAgent {
           ```
           and a body containing `<sasToken>`.]*/
             let amqpMessage = new AmqpMessage();
-            amqpMessage.applicationProperties = {
+            amqpMessage.application_properties = {
               operation: 'put-token',
               type: 'servicebus.windows.net:sastoken',
               name: audience
             };
             amqpMessage.body = token;
-            amqpMessage.properties = {
-              to: '$cbs',
-              messageId: uuid.v4(),
-              reply_to: 'cbs'
-            };
+            amqpMessage.to = '$cbs';
+            //
+            // Just as a reminder:  For cbs, the message id and the correlation id
+            // always stayed as string values.  They never went on over the wire
+            // as the amqp uuid type.
+            //
```

Comment:
minor nit: the way this comment is phrased ("Just as a reminder", "never went over the wire") refers to tribal knowledge and seem to indicate a context that someone in the future won't have. I would suggest simply stating the fact/protocol: "For CBS, the message id and correlation id are encoded as string".

Reply:
Well this is where the tribe is being told the story.  This of course has never been well specified.  I will restate with the far less chatty: For CBS, the message id and correlation id are encoded as string.

---
In reply to: [209348680](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209348680) [](ancestors = 209348680)